packages/license-scanner/src/getPkgInfo.ts

```diff
@@ -0,0 +1,354 @@
+ * @param field the string to be converted
+ * @returns string | null
+ */
+function coerceToString (field: unknown): string | null {
+  const string = String(field)
+  return typeof field === 'string' || field === string ? string : null
+}
+
+/**
+ * Parses the value of the license-property of a
+ * package manifest and return it as a string
+ * @param field the value to parse
+ * @returns string
+ */
+function parseLicenseManifestField (field: unknown) {
+  if (Array.isArray(field)) {
+    const licenses = field
+    const licenseTypes = licenses.reduce((listOfLicenseTypes, license) => {
+      const type = coerceToString(license.type) ?? coerceToString(license.name)
+      if (type) {
+        listOfLicenseTypes.push(type)
+      }
+      return listOfLicenseTypes
+    }, [])
+
+    if (licenseTypes.length > 1) {
+      const combinedLicenseTypes = licenseTypes.join(' OR ') as string
+      return `(${combinedLicenseTypes})`
+    }
+
+    return licenseTypes[0] ?? null
+  } else {
+    return (field as { type: string })?.type ?? coerceToString(field)
+  }
+}
+
+/**
+ * Reads the license field or LICENSE file from
+ * the directory of the given package manifest
+ *
+ * If the package.json file is missing the `license`-property
+ * the root of the manifest directory will be scanned for
+ * files named listed in the array LICENSE_FILES and the
+ * contents will be returned.
+ *
+ * @param {*} pkg the package to check
+ * @param {*} opts the options for parsing licenses
+ * @returns Promise<LicenseInfo>
+ */
+async function parseLicense (
+  pkg: {
+    manifest: PackageManifest
+    files:
+    | { local: true, files: Record<string, string> }
+    | { local: false, files: Record<string, PackageFileInfo> }
+  },
+  opts: { cafsDir: string }
+): Promise<LicenseInfo> {
+  let licenseField: unknown = pkg.manifest.license
+  if ('licenses' in pkg.manifest) {
+    licenseField = (
+      pkg.manifest as PackageManifest & {
+        licenses: unknown
+      }
+    ).licenses
+  }
+  const license = parseLicenseManifestField(licenseField)
+
+  // check if we discovered a license, if not attempt to parse the LICENSE file
+  if (!license || /see license/i.test(license)) {
+    const { files: pkgFileIndex } = pkg.files
+    for (const filename of LICENSE_FILES) {
+      // check if the a file with the expected name exists in the file index
+      if (!(filename in pkgFileIndex)) {
+        continue
+      }
+
+      const licensePackageFileInfo = pkgFileIndex[filename]
+      let licenseContents: Buffer | undefined
+      if (pkg.files.local) {
+        licenseContents = await readFile(licensePackageFileInfo as string)
+      } else {
+        licenseContents = await readLicenseFileFromCafs(
+          opts.cafsDir,
+          (licensePackageFileInfo as PackageFileInfo).integrity
+        )
+      }
+
+      return {
+        name: 'Unknown',
+        licenseFile: licenseContents?.toString('utf-8'),
+      }
```

Comment:
What if the package has multiple license files?

Reply:
I have seen such packages in the past.

Reply:
Packages that don't use `license`-property but have multiple license files? Happy to improve the code after this PR has been merged sounds like a pretty edge case to me

Reply:
I work in a company that archives npm data and I can tell you without question that there are more packages that do that than you'd think. Roughly 10% of packages in the registry. People do nutty stuff in package.json and npm has no validation.

Reply:
If you happen to archive npm data via work, could younbe so kind and suggest some npm packages that does this so I can make it part of a test case?

Reply:
give me a bit of time to get to that, but I'll take a look for you. if I don't get to this in a week, please ping me

Reply:
Sure, of course,  I am already happy you want to take the time help me :)